uwblueprint/supportive-housing
# Connecting tags when creating, filtering and editing log record

## Notion task link


[Task](https://www.notion.so/uwblueprintexecs/Connecting-Tags-To-Log-Records-c0ba6affff9646068320177bbef67c46)



## Implementation description
* Enable multi-select tags input when the user creates, edits or filters a log record
* Make GET request to backend to get all ACTIVE tags
* Change backend logic to use tag id (rather than tag name) when creating or filtering a log record



## Steps to test
1. At the moment, we don't have any way to create a tag from the frontend or backend, so you'll need to manually insert it into the db
```
insert into tags (tag_id, name, status) values (1, 'Test Tag', 'Active');
```
2. On the frontend, when creating a log record, click the "Tags" input box and check that the dropdown displays all tags that are currently in the tags table in the db
3. Add a couple of log records with: a) No tags, b) One tag, c) Multiple tags (make sure that you get a success response on the frontend and check that the log_record_tag table has also been correctly updated)
4. Try editing log records and updating their tags section: a) Remove all tags, b) Add new tags c) Don't change tags (similar to create, make sure this gives successful response and updates log_record_tag table accordingly)
5. Filter by a single tag or multiple tags and ensure that all log records with ALL of the specified tags are being displayed (also check that the count of log records is correct)



## What should reviewers focus on?
* Let me know if I should split this into multiple PRs (backend and frontend) to make it easier to review


## Checklist
